src/rust/engine/src/nodes.rs

```diff
@@ -1205,13 +1209,23 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let span_id = generate_random_64bit_string();
-    let node_workunit_params = if context.session.should_record_zipkin_spans() {
-      let node_name = format!("{}", self);
-      let start_time = std::time::SystemTime::now();
-      Some((node_name, start_time, span_id.clone()))
-    } else {
-      None
+
+    let handle_workunits =
+      context.session.should_report_workunits() || context.session.should_record_zipkin_spans();
+
+    let maybe_display_info: Option<String> = match self {
+      NodeKey::Task(ref task) => task.get_display_info().map(|s| s.to_owned()),
```

Comment:
This block is rendering/cloning strings, and should not run if `!handle_workunits`.